packages/react/src/transactions/useFollow.ts

```diff
@@ -24,82 +24,69 @@ import {
 
 import { Session, SessionType, useSession } from '../authentication';
 import { UseDeferredTask, useDeferredTask } from '../helpers/tasks';
-import { useLazyModuleMetadata } from '../misc';
 import { AsyncTransactionResult } from './adapters/AsyncTransactionResult';
 import { useFollowController } from './adapters/useFollowController';
 
 export class PrematureFollowError extends Error {
   name = 'PrematureFollowError' as const;
 }
 
-function useFollowRequestFactory() {
-  const { execute: fetchModule } = useLazyModuleMetadata();
-
-  return async (args: FollowArgs, session?: Session): Promise<FollowRequest> => {
-    invariant(
-      session?.authenticated,
-      'You must be authenticated to use this operation. Use `useLogin` hook to authenticate.',
-    );
-    invariant(
-      session.type === SessionType.WithProfile,
-      'You must have a profile to use this operation.',
-    );
-
-    const followPolicy = resolveFollowPolicy(args.profile);
-
-    switch (followPolicy.type) {
-      case FollowPolicyType.CHARGE:
-        return {
-          kind: TransactionKind.FOLLOW_PROFILE,
-          fee: {
-            amount: followPolicy.amount,
-            contractAddress: followPolicy.contractAddress,
-            recipient: followPolicy.recipient,
-          },
-          profileId: args.profile.id,
-          signless: false,
-          sponsored: args.sponsored ?? true,
-        };
-
-      case FollowPolicyType.ANYONE:
-        return {
-          kind: TransactionKind.FOLLOW_PROFILE,
-          profileId: args.profile.id,
-          signless: session.profile.signless,
-          sponsored: args.sponsored ?? true,
-        };
+/**
+ * @internal
+ */
+export function createFollowRequest(args: FollowArgs, session?: Session): FollowRequest {
```

Comment:
does it need to be exported? I don't see it used anywhere outside of this file

Reply:
no more, cleared some of this in the following PR.